src/modules/EventModule.cpp

```diff
@@ -0,0 +1,463 @@
+					}
+
+					for (auto& role : roles)
+					{
+						m_EventWhitelist[guild].removeOne(role);
+					}
+
+					UmikoBot::Instance().createMessage(channel.id(), "**The roles have been removed.**\nPeople with the role(s) can no longer launch or end events!");
+
+				});
+		});
+	RegisterCommand(Commands::EVENT_LAUNCH, "launch", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+
+		client.getGuildMember(channel.guildId(), message.author().id())
+			.then([=](const Discord::GuildMember& member)
+			{
+					bool found = false;
+
+					for (auto& allowedRole : m_EventWhitelist[channel.guildId()])
+					{
+						if (member.roles().contains(allowedRole))
+						{
+							found = true;
+							break;
+						}
+					}
+
+					if (!found)
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**You don't have permissions to launch events.**");
+						return;
+					}
+
+					auto& config = getEventServerData(channel.guildId());
+					CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+					auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+
+					if (args.size() != 3)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+						return;
+					}
+
+					if (config.isEventRunning)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "An event is already running. Please stop that to start a new event!");
+						return;
+					}
+
+					double time = args.at(2).toDouble();
+
+					QRegExp re("[+]?\\d*\\.?\\d+");
+					if (!re.exactMatch(args.at(2)))
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Wrong Usage of Command!**.");
+						return;
+					}
+					if (time > 24)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Currently you can't launch events with an expiry time of more than one day**.");
+						return;
+					}
+					if (config.eventTimer != nullptr) //Delete the previous timer
+					{
+						delete config.eventTimer;
+						config.eventTimer = nullptr;
+					}
+					config.eventTimer = new QTimer;
+
+					config.eventTimer->setInterval(time * 3600000);
+					config.eventTimer->setSingleShot(true);
+
+					if (args.at(1) == "HighRiskHighReward")
+					{
+						QString num = QString::number(time);
+
+						Discord::Embed embed;
+						embed.setColor(15844367);
+						embed.setTitle("**HighRiskHighReward** event has been launched for `" + num + "` hour(s)!");
+						embed.setDescription("Give command `!event` to see what event is running and its changes!");
+
+						config.isEventRunning = true;
+						config.eventHighRiskHighRewardRunning = true;
+						currencyConfig.stealSuccessChance -= highRiskRewardStealDecrease;
+						UmikoBot::Instance().createMessage(message.channelId(), embed);
+					}
+					if (args.at(1) == "LowRiskLowReward")
```

Comment:
What if neither of these are true?

Reply:
Hmm gotta add that, thanks for pointing that out! 😄